src/components/ThemeItemList.tsx

```diff
@@ -0,0 +1,168 @@
+  width: 100%;
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  gap: 24px 8px;
+`;
+const linkStyle = css`
+  text-decoration: none;
+  color: inherit;
+  background-color: transparent;
+  cursor: pointer;
+`;
+const itemBox = css`
+  width: 100%;
+`;
+const itemImgStyle = css`
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  border-radius: 4px;
+  background-color: ${theme.colors.backgroundDisabled};
+`;
+const brandP = css`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  font-size: ${theme.typography.subtitle2Regular.fontSize};
+  font-weight: ${theme.typography.subtitle2Regular.fontWeight};
+  line-height: ${theme.typography.subtitle2Regular.lineHeight};
+  color: ${theme.colors.textSub};
+  margin: 0;
+  text-align: left;
+  display: -webkit-box;
+  -webkit-line-clamp: 2;
+  -webkit-box-orient: vertical;
+`;
+const itemNameH6 = css`
+  font-size: ${theme.typography.subtitle2Regular.fontSize};
+  font-weight: ${theme.typography.subtitle2Regular.fontWeight};
+  line-height: ${theme.typography.subtitle2Regular.lineHeight};
+  color: ${theme.colors.textDefault};
+  margin: 0;
+  text-align: left;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: -webkit-box;
+  -webkit-line-clamp: 2;
+  -webkit-box-orient: vertical;
+`;
+const priceP = css`
+  font-size: ${theme.typography.title2Bold.fontSize};
+  font-weight: ${theme.typography.title2Bold.fontWeight};
+  line-height: ${theme.typography.title2Bold.lineHeight};
+  color: ${theme.colors.textDefault};
+  margin: 0;
+  text-align: left;
+  word-break: break-word;
+`;
+const spacer12 = css`
+  height: ${theme.spacing.spacing3};
+`;
+const spacer4 = css`
+  height: ${theme.spacing.spacing1};
+`;
+const noItemsDiv = css`
+  width: 100%;
+  height: 240px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+const noItemsText = css`
+  font-size: ${theme.typography.subtitle2Regular.fontSize};
+  font-weight: ${theme.typography.subtitle2Regular.fontWeight};
+  line-height: ${theme.typography.subtitle2Regular.lineHeight};
+  color: ${theme.colors.textDefault};
+  margin: 0;
+  width: 100%;
+  text-align: center;
+`;
+const loadingStyle = css`
+  width: 100%;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+const loadingGifStyle = css`
+  width: 50px;
+`;
+const bottomLoading = css`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 24px 0;
+`;
+
+const ThemeItemList = () => {
+  const { themeId } = useParams<{ themeId: string }>();
+  const { items, loading, hasMore, lastRef } = useThemeItemList(themeId);
+
+  const { navigateIfLoggedIn } = useAuthNavigation();
+
+  const goOrder = (id: number) => {
+    navigateIfLoggedIn(`/order/${id}`);
+  };
+
+  if (loading && items.length === 0)
+    return (
+      <div css={loadingStyle}>
+        <img css={loadingGifStyle} src={loadingGif} alt="Loading..." />
+      </div>
+    );
```

Comment:
요런 로딩 컴포넌트는 따로 분리해서 사용할 수 있다면 재상용성이나 가독성이 더 좋겠네용! 

```suggestion
  if (loading && items.length === 0)
    return <Loading />;
```